enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+capabilities they can enable/disable when configuring the installation.
+
+Note: Resources are enabled unless explicitly excluded by configuration.  In the future we may allow
+resources to be excluded unless explicitly included, this can be done by introducing a new annotation
+such as `capability.openshift.io/defaultDisable`.  If we do so, we will need to define how that choice
+interacts with the cluster scoped "default capabilities on/off" configuration setting.  For now, this
+EP takes the position that individual components that exist in the payload cannot choose to be
+default-disabled.
+
+2) Defining an [install config api](https://github.com/openshift/installer/blob/790048166067273d34f76bea4220fa395b1cce1b/pkg/types/installconfig.go#L70) field whereby the user can opt out of specific capabilities.
+
+```yaml
+InstallConfig.capabilities
+  inclusionDefault: [Include|Exclude]
+  exclude:
+  - console
+  - samples
+  include:
+  - registry
+```
+
+Which resources ultimately get installed for a given cluster would be the set of resources encompassed
+by the CLUSTER_PROFILE(if any), minus any resources explicitly excluded by the excluded capabilities configuration,
+plus any explicitly included capabilities.  Any resource that is not excluded by the CLUSTER_PROFILE, is not
+included via the Include stanza, and is not Excluded via the Exclude stanza, will be filtered or not based on
+the `inclusionDefault` value (Include or Exclude).  The `inclusionDefault` field itself will default to `Include`.
+
+The `inclusionDefault` field (carried through to the CVO) allows an Admin to choose what behavior they want for new
+capabilities that might come as part of a future upgrade (either have the new capability installed during
+the upgrade, or exclude it)
+
+Specifying the same capability as an `include` and an `exclude` is an error condition.
+
+
+Examples of candidate components to be treated as capabilities:
+
+* console
+* imageregistry
+* samples
+* cluster baremetal operator
+* olm/marketplace
+* csi-*
+* insights
+* monitoring
+* ???
+
+
+
+3) Pass the list of filtered annotations to the CVO.  This is distinct from overrides because overrides
+put the cluster in an unsupported state.  Filtered annotations are supported for upgrades.  The filtered
+components will be listed in the [ClusterVersion](https://github.com/openshift/api/blob/4436dc8be01e8dcd8b250e1b32bb0fbd64ba78ac/config/v1/types_cluster_version.go#L35) object:
+
+```yaml
+  spec:
+    capabilities:
+      inclusionDefault: [Include|Exclude]
+      exclude:
+      - console
+      - samples
+      include:
+      - registry
+```
+
+The CVO will filter out(not apply/reconcile) resources that are annotated with key/value
+`capability.openshift.io/name=$exclusions` and listed in the `exclude` section, or are not listed
+and the `default` setting is `exclude`.  In addition any explicitly included capability
+should be added, regardless of the CLUSTER_PROFILE.
```

Comment:
We should link the cluster profile enhancement here i.e. https://github.com/openshift/enhancements/blob/7da960b98b2835022823f569efa1e18e9e410696/enhancements/update/cluster-profiles.md

Reply:
adding